### DISCUSSION TITLE
High idle load (fixed in 0.145.6)

### DISCUSSION THREAD

**Discussion by InB4DevOps** [lloesche/valheim-server-docker#10] [Q&A] [answered]
Hey there, 

I have this server running on my Synology NAS (DS416play w/ 8GB RAM) and while it runs surprisingly well I'm wondering if you guys are also having a rather high CPU consumption when no one is actually connected to the server? 

I'm seeing a constant 33%-40% load for the valheim server process..

Thanks

**Comment by lloesche** [comment, marked as answer]
I can confirm this. I'm running Valheim server on a 2.4 GHz 6 Core Intel Xeon E5-2620 and the Valheim server process is constantly consuming ~130% CPU (i.e. 1.3 Cores).

```
  PID USER      PR  NI    VIRT    RES    SHR S  %CPU  %MEM     TIME+ COMMAND
   92 steam     20   0 8966868   1.7g  50140 S 129.3   2.7  28:11.47 valheim_server.
```

And at the moment nobody is connected to the server.

**Reply by InB4DevOps** [reply]
> I can confirm this. I'm running Valheim server on a 2.4 GHz 6 Core Intel Xeon E5-2620 and the Valheim server process is constantly consuming ~130% CPU (i.e. 1.3 Cores).
> 
> ```
>   PID USER      PR  NI    VIRT    RES    SHR S  %CPU  %MEM     TIME+ COMMAND
>    92 steam     20   0 8966868   1.7g  50140 S 129.3   2.7  28:11.47 valheim_server.
> ```
> 
> And at the moment nobody is connected to the server.

Only now did I notice this detail. How do you have the server process running as user steam? Is this without using the container?
Mine runs as root.

![image](https://user-images.githubusercontent.com/73686890/108200323-9b190d80-711e-11eb-99e7-e479e280d0fa.png)

**Comment by InB4DevOps** [comment]
Look at https://valheim.fandom.com/wiki/Change_Log/Patch_Notes

Seems they have fixed it in the latest update.

`Dedicated server CPU usage fix (Framerate limiter)`

EDIT: Just installed it and the server load in general is much lower. (both with players connected and idle)

**Comment by LHolst** [comment]
```
NAME                CPU %               MEM USAGE / LIMIT    MEM %               NET I/O             BLOCK I/O           PIDS
valheim-server      38.71%              1.691GiB / 7.71GiB   21.93%              4.84GB / 2.1GB      319GB / 1.01GB      40
```

`on idle/ no connections` after `updater` did his job.

**Reply by lloesche** [reply]
Yup can confirm
```
  PID USER      PR  NI    VIRT    RES    SHR S  %CPU  %MEM     TIME+ COMMAND
22542 root      20   0 9214320   2.2g  49292 S  26.9   3.5  88:37.63 valheim_server.
```

**Comment by schildbach** [comment]
I also have a constant 20-30% CPU load with no players connected.

Is this something the devs are aware of, or did they state wether it could be fixed?

Valheim server: 0.211.11
Docker image: latest
Host system: 4 virtual AMD EPYC 7282 cores @ 2800 MHz

**Comment by sirber** [comment]
still the same in 2025

<img width="885" height="162" alt="image" src="https://github.com/user-attachments/assets/339c3f56-9b20-4ad9-8df3-492f35d8d22e" />
